File: Problem/Package/Task/150504/PEARL.pas
program Zerolife;
var f,g:text;
    i,t,dem:longint;
    d,c:int64;
function kt(n:int64):boolean;
var s:longint;
    sum:array[1..1458] of longint;
begin
 s:=0;
 while n>0 do
  begin
   s:=s+sqr((n mod 10));
   n:=n div 10;
  end;
 n:=s;
 fillchar(sum,sizeof(sum),0);
 while sum[n]<=1 do
  begin
   s:=0;
   while n>0 do
    begin
     s:=s+sqr((n mod 10));
     n:=n div 10;
    end;
   if s=1 then exit(false) else inc(sum[s]);
   n:=s;
  end;
 exit(true);
end;
begin
 assign(f,'PEARL.INP'); reset(f);
 assign(g,'PEARL.OUT'); rewrite(g);
 readln(f,t);
 for i:=1 to t do
  begin
   readln(f,d,c); dem:=0;
   while d<=c do
    begin
     if kt(d) then inc(dem); //else writeln(g,d);
     inc(d);
    end;
   writeln(g,dem);
  end;
 close(f);
 close(g);
end.

File: Problem/Package/Task/TB.Trinh/pearl.cpp
#include <bits/stdc++.h>

using namespace std;
ifstream fi ("pearl.inp");
ofstream fo ("pearl.out");
int t;
long long l,r;
int d[1500];
long long check(long long x)
{
    int sum=0;
    long long k=x;
    while (k>0)
    {
        int v=k%10;
        sum=sum+v*v;
        k=k/10;
    }
    if(sum==1) return 1;
    if(d[sum]==1) return 0;
    return sum;
}
int main()
{
    fi>>t;
    for (int p=1;p<=t;p++)
    {
        long long l,r;
       int dem=0;
       fi>>l>>r;
       for (long long i=l;i<=r;i++)
            {
                for (int i=1;i<=1500;i++)
                    d[i]=0;
                if(i<1500) d[i]=1;
                long long a=i;
                while(1)
                {
                    a=check(a);
                    if(a==0) {dem++;break;}
                    else d[a]=1;
                    if(a==1) break;
                }
            }
            fo<<dem<<endl;
    }
    return 0;
}
